### PR TITLE
web: improve warning display in overview

### PR DESCRIPTION
**Pull request by nicks**
Hello @lizzthabet,

Please review the following commits I made in branch nicks/warnings2:

b5374ce8b54fdb305a01f648ce335ef4464ace47 (2021-09-07 17:07:19 -0400)
web: improve warning display in overview

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
Here's what it looks like before:
https://storybook.tilt.dev/?path=/story/legacy-ui-sidebaritemview--one-item-warning
Screenshot from 2021-09-07 15-52-47

Here's what it looks like after:
Screenshot from 2021-09-07 16-49-36
Screenshot from 2021-09-07 16-49-18

File: web/src/OverviewTableStatus.tsx [hunk 1]
```diff
@@ -98,20 +98,28 @@ export default function OverviewTableStatus(props: OverviewTableStatusProps) {
       msg = isBuild ? "Update Pending" : "Runtime Pending"
       classes = "is-pending"
       break
-    case ResourceStatus.Warning:
+
+    case ResourceStatus.Warning: {
+      let buildDurText = lastBuildDur
+        ? ` in ${formatBuildDuration(lastBuildDur)}`
+        : ""
+      icon = <WarningSvg fill={Color.yellow} width="10px" height="10px" />
```
Comment (lizzthabet) [inline web/src/OverviewTableStatus.tsx:106]: i noticed that the icon shrinks a ton when the table shrinks. adding `flex-shrink: 0;` to the `StatusIcon` class fixed it!

File: web/src/OverviewTable.stories.tsx [hunk 2]
```diff
@@ -44,6 +47,32 @@ export default {
 
 export const TwoResources = () => <OverviewTable view={twoResourceView()} />
 
+export const TiltfileWarning = () => {
+  let view = nResourceView(1)
```
Comment (lizzthabet) [inline web/src/OverviewTable.stories.tsx:51]: (nitpick / optional) i found myself curious about how a resource with warnings would look with other resources around it.  
```suggestion
  let view = nResourceView(10)
```

**Review by lizzthabet — APPROVED** [review]
lgtm! i'm not sure if it's worth fiddling with the column widths to fit the `status` column on one line.

Screen Shot 2021-09-08 at 12 59 38 PM

eh? i changed `ResourceName` to 250px, `Type` to 130px, and `Status` to 250px.
Screen Shot 2021-09-08 at 1 04 38 PM

File: web/src/OverviewTable.stories.tsx (hunk 2 above)
Comment (nicks) [reply web/src/OverviewTable.stories.tsx]: done

File: web/src/OverviewTableStatus.tsx (hunk 1 above)
Comment (nicks) [reply web/src/OverviewTableStatus.tsx:107]: done

**Review by nicks — COMMENTED** [review]
re: table width - eh, i just removed the "with issues" text, which is probably redundant with the icon anyway.

**Pull request merged**